external-crates/move/crates/move-analyzer/src/symbols.rs

```diff
@@ -997,63 +1007,119 @@ pub fn get_symbols(
+                    if let Ok(lib) = res {
+                        eprintln!("created pre-compiled libs for {:?}", pkg_path);
+                        let res = Arc::new(lib);
+                        pkg_deps.insert(pkg_path.to_path_buf(), res.clone());
+                        Some(res)
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                }
             }
         };
-        eprintln!("compiled to typed AST");
-        let (compiler, typed_program) = compiler.into_ast();
-        typed_ast = Some(typed_program.clone());
-
-        // compile to bytecode for accurate diags
-        eprintln!("compiling to bytecode");
-        let compilation_result = compiler.at_typing(typed_program).build();
-        let (units, diags) = match compilation_result {
-            Ok(v) => v,
-            Err((_pass, diags)) => {
+        if compiled_deps.is_some() {
+            // if successful, remove only source deps but keep bytecode deps as they
+            // were not used to construct pre-compiled lib in the first place
+            dependencies.remove_deps(src_names);
+        }
+        compiled_deps
+    } else {
+        None
+    };
+
+    build_plan.compile_with_driver_and_deps(
+        dependencies,
+        &mut std::io::sink(),
+        |mut compiler| {
+            compiler = compiler.set_vfs_root(overlay_fs_root.clone());
+            compiler = compiler.set_pre_compiled_lib_opt(compiled_libs.clone());
+            // extract expansion AST
+            let (files, compilation_result) = compiler.run::<PASS_PARSER>()?;
+            let (_, compiler) = match compilation_result {
+                Ok(v) => v,
+                Err((_pass, diags)) => {
+                    let failure = true;
+                    diagnostics = Some((diags, failure));
+                    eprintln!("parsed AST compilation failed");
+                    return Ok((files, vec![]));
+                }
+            };
+            eprintln!("compiled to parsed AST");
+            let (compiler, parsed_program) = compiler.into_ast();
+            parsed_ast = Some(parsed_program.clone());
+
+            // extract typed AST
+            let compilation_result = compiler.at_parser(parsed_program).run::<PASS_TYPING>();
+            let compiler = match compilation_result {
+                Ok(v) => v,
+                Err((_pass, diags)) => {
+                    let failure = true;
+                    diagnostics = Some((diags, failure));
+                    eprintln!("typed AST compilation failed");
+                    return Ok((files, vec![]));
+                }
+            };
+            eprintln!("compiled to typed AST");
+            let (compiler, typed_program) = compiler.into_ast();
+            typed_ast = Some(typed_program.clone());
+
+            // compile to bytecode for accurate diags
```

Comment:
Couldn't we just stop at CFGIR? is there any reason to continue to bytecode?

Reply:
I assumed that diags can be generated at every pass, including bytecode translation. Changed to finish at CFGIR compilation.